memorious/operations/aleph.py

```diff
@@ -132,6 +136,52 @@ def aleph_folder(context, data):
             backoff(ae, try_number)
 
 
+def aleph_emit_entity(context, data):
+    api = get_api(context)
+    if api is None:
+        return
+    collection_id = get_collection_id(context, api)
+    entity_id = data.get("entity_id")
+    source_url = data.get("source_url", data.get("url"))
+    foreign_id = data.get("foreign_id", data.get("request_id", source_url))
+    # Fetch id from cache
+    cached_key = context.get_tag(make_key(collection_id, foreign_id, entity_id))
+    if cached_key:
+        context.log.info("Skip aleph upload: %s", foreign_id)
+        data["aleph_id"] = cached_key
+        context.emit(data=data, optional=True)
+        return
+    else:
+        entity_id = uuid.uuid4().hex
+
+    for try_number in range(api.retries):
+        rate = settings.MEMORIOUS_RATE_LIMIT
+        rate_limit = get_rate_limit("aleph", limit=rate)
+        rate_limit.comply()
+        try:
+            api.write_entity(
+                collection_id,
+                {
+                    "schema": data.get("schema"),
+                    "properties": data.get("properties"),
+                },
+                entity_id,
+            )
+
+            context.log.info("Aleph document entity ID: %s", entity_id)
+            # Save the document id in cache for future use
+            context.set_tag(make_key(collection_id, foreign_id, entity_id), entity_id)
+            data["aleph_id"] = entity_id
```

Comment:
This is going to produce incorrect behaviour.

The signed key is going to be different from the supplied key. eg: if we upload an entity with the id `banana`, the actual key produced by aleph will be something like `banana.123abc`. We want to store the signed key as the `aleph_id`. Otherwise any reference to the entity made by using the unsigned id `banana` won't resolve.

See for example how we extract the id of a new document from the response in `aleph_emit_document`

`write_entity` doesn't return the response though. So we'll need to make that change in alephclient.